src/converter.ts

```diff
@@ -515,7 +515,7 @@ export class ProtocolToMonacoConverter {
 
     asMarker(diagnostic: Diagnostic): monaco.editor.IMarkerData {
         return {
-            code: diagnostic.code as string,
+            code: typeof diagnostic.code === "number" ? diagnostic.code.toString() : diagnostic.code as string,
```

Comment:
You're right, it can. I just assumed that tsc would complain about this because it's a `string | number` but it didn't.